sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -112,28 +112,62 @@ def __exit__(self, *args):
 
     @property
     def url(self):
+        """The full endpoint URL to this entity, including SAS token if used.
+
+        This could be either the primary endpoint,
```

Comment:
nit: (here and elsewhere) line is unnecessarily short

Reply:
for this one, I prefer personally to cut at the comma :)